projects/packages/jitm/src/js/jetpack-jitm.js

```diff
@@ -248,17 +248,31 @@ jQuery( document ).ready( function ( $ ) {
 				}
 
 				// for now, always take the first response
-				setJITMContent( $el, response[ 0 ], redirect );
+				const data = response [ 0 ];
+
+				const LICENSE_ACTIVATION_NOTICE_KEYS = [
+					'single_licensing_activation_notice',
+					'multiple_licensing_activation_notice',
+				];
+
+				const inJPLicenseActivationPage = window?.location?.href.indexOf( 'jetpack#/add-license' ) >= 0;
+				const isJPLicenseActivationNotice = LICENSE_ACTIVATION_NOTICE_KEYS.includes( data.id );
+
+				// We do not want to display License Activation JITM in the activation page
+				if ( !( isJPLicenseActivationNotice && inJPLicenseActivationPage) ) {
+					setJITMContent( $el, data, redirect );
+				}
```

Comment:
I see for Jetpack's own React admin UI we have some code above that includes the hash of the page in the query sent to the API
https://github.com/Automattic/jetpack/blob/50838340dc0d32459c1a21a070995b690d3e41cd/projects/packages/jitm/src/js/jetpack-jitm.js#L226-L231
(originally added in #10889 for a similar reason)

Could we do something similar for my-jetpack so you could filter the JITMs on the server side instead?

Reply:
@anomiex thanks for pointing me to the right direction. I had some doubts adding this logic here but yes doing it in the server side is the better approach. It makes more sense. I'll have a look.. 👍

Reply:
Hi @anomiex, Sorry for the delay in getting back to work on this PR. I changed the **jetpack-jitm.js** file so that the hash is included in the message path.I made sure it only applied to the "My Jetpack" pages and did not break anything.

The current backend logic already filters the new message path with a hash, so no changes are needed.

Reply:
Thanks! Code looks good to me at first glance. Once your team reviews for functionality, tag it with "[Status] Needs Review" and I or someone else from Crew will have another look.